Add docs about the CLI (#6831)
huggingface/datasets: docs/source/cli.mdx

# Command Line Interface (CLI)

🤗 Datasets provides a command line interface (CLI) with useful shell commands to interact with your dataset.

You can check the available commands:
```shell
$ datasets-cli --help
usage: datasets-cli <command> [<args>]

positional arguments:
  {convert,env,test,run_beam,dummy_data,convert_to_parquet}
                        datasets-cli command helpers
    convert             Convert a TensorFlow Datasets dataset to a HuggingFace Datasets dataset.
    env                 Print relevant system environment info.
    test                Test dataset implementation.
    run_beam            Run a Beam dataset processing pipeline
    dummy_data          Generate dummy data.
    convert_to_parquet  Convert dataset to Parquet

optional arguments:
  -h, --help            show this help message and exit
```

## Convert to Parquet

> Easily convert your Hub script-dataset to Parquet files, so that the dataset viewer will be supported.

Reviewer: ```suggestion
Easily convert your Hub [script-based dataset](https://huggingface.co/docs/datasets/dataset_script) to [data-only](https://huggingface.co/docs/hub/datasets-manual-configuration), so that the dataset viewer will be supported.
```

```shell
$ datasets-cli convert_to_parquet --help
usage: datasets-cli <command> [<args>] convert_to_parquet [-h] [--token TOKEN] [--revision REVISION] [--trust_remote_code] dataset_id

positional arguments:

>   dataset_id           source dataset ID

Reviewer: should we precise what a dataset ID is? Here, or inside a small example usage of the CLI below?

Reviewer: I feel like people like to copy / edit / paste (I do)

Author: I have added an example, because the content here is what the CLI command --help outputs; maybe I could add the example to the CLI command --help as well.

Author: Done.

```shell
$ datasets-cli convert_to_parquet --help
usage: datasets-cli <command> [<args>] convert_to_parquet [-h] [--token TOKEN] [--revision REVISION] [--trust_remote_code] dataset_id

positional arguments:
  dataset_id           source dataset ID, e.g. USERNAME/DATASET_NAME or ORGANIZATION/DATASET_NAME

optional arguments:
  -h, --help           show this help message and exit
  --token TOKEN        access token to the Hugging Face Hub
  --revision REVISION  source revision
  --trust_remote_code  whether to trust the code execution of the load script
```

This command:
- makes a copy of the script on the "main" branch into a dedicated branch called "script" (if it does not already exists)
- creates a pull request to the Hub dataset to convert it to Parquet files (and deletes the script from the main branch)

If in the future you need to recreate the Parquet files from the "script" branch, pass the `--revision script` argument.

Note that you should pass the `--trust_remote_code` argument only if you trust the remote code to be executed locally on your machine.

For example:
```shell

> $ datasets-cli convert_to_parquet USERNAME/DATASET_NAME --token USER_ACCESS_TOKEN

Reviewer: Let's just remove the `$`

```suggestion
datasets-cli convert_to_parquet USERNAME/DATASET_NAME --token USER_ACCESS_TOKEN
```

Author: OK, I added to make clear what the input was (and the output, without leading dollar sign). But I will remove it then.

Author: Is it OK if I use `>>>`, as huggingface-hub docs do? https://huggingface.co/docs/huggingface_hub/en/guides/cli

That way, in the examples above (with input and output), the copy/paste just copies the input. If I remove any leading symbol, the copy/paste will copy input and output...

Reviewer: Hmmm, it seems like a bug to use `>>>` for cli @Wauplin 

In our case we have no output, so I would just remove `$`

Author: We have output in the 2 code snippets above the third on.

Reviewer: Ah sure. Hmmm, let's use `>>>` then and see the alternative in another PR maybe

Author: See:
- https://github.com/huggingface/datasets/pull/6831/files#diff-25d866ca4099e701561fffdb962611aedbc2598e7fc12975831cc614277f10f2R6-R22
- https://github.com/huggingface/datasets/pull/6831/files#diff-25d866ca4099e701561fffdb962611aedbc2598e7fc12975831cc614277f10f2R28-R40

Reviewer 2: > Hmmm, it seems like a bug to use >>> for cli

> Ah sure. Hmmm, let's use >>> then and see the alternative in another PR maybe

Happy to align to whatever convention we want. I haven't really thought the it TBH

Author: The same for me.

> >>> datasets-cli convert_to_parquet USERNAME/DATASET_NAME --token USER_ACCESS_TOKEN

Reviewer 2: ```suggestion
>>> datasets-cli convert_to_parquet USERNAME/DATASET_NAME
```

I would not advertise the `--token` arg in the example as this shouldn't be the recommended way (best to login with env variable or `huggingface-cli login`)

Author: Thanks for your comment @Wauplin. I am fixing that.

Author: Fixed by:
- #6839
